src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -346,60 +369,96 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
   }, [])
 
   const handleSubmit = useCallback(() => {
-    if (!(stakingAssetAccountId && runeAddress && isValidStakingAmount)) return
+    if (!(assetAccountId && stakingAssetAccountId && runeAddress && isValidStakingAmount)) return
 
-    setConfirmedQuote({
+    const _confirmedQuote = {
       stakingAssetAccountId,
       stakingAssetId,
-      stakingAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, stakingAsset?.precision ?? 0),
-
+      stakingAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, asset?.precision ?? 0),
       runeAddress,
-    })
+    }
+
+    setConfirmedQuote(_confirmedQuote)
+
+    if (isBridgeRequired) {
+      const bridgeQuote: RfoxBridgeQuote = {
+        sellAssetId: assetId,
+        buyAssetId: stakingAssetId,
+        bridgeAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, asset?.precision ?? 0),
+        sellAssetAccountId: assetAccountId,
+        buyAssetAccountId: stakingAssetAccountId,
+      }
+      return history.push({ pathname: BridgeRoutePaths.Confirm, state: bridgeQuote })
+    }
+
     history.push(StakeRoutePaths.Confirm)
   }, [
-    stakingAsset?.precision,
-    amountCryptoPrecision,
-    history,
-    isValidStakingAmount,
+    assetAccountId,
     runeAddress,
+    isValidStakingAmount,
+    stakingAssetId,
+    amountCryptoPrecision,
+    asset?.precision,
     setConfirmedQuote,
+    isBridgeRequired,
+    history,
+    assetId,
     stakingAssetAccountId,
-    stakingAssetId,
   ])
 
+  const buyAssetSearch = useModal('buyAssetSearch')
+
+  const handleStakingAssetClick = useCallback(() => {
+    if (!(stakingAsset && l1Asset)) return
+
+    buyAssetSearch.open({
+      onAssetClick: asset => setAssetId(asset.assetId),
+      title: 'common.selectAsset',
+      assets: [stakingAsset, l1Asset],
+    })
+  }, [stakingAsset, l1Asset, buyAssetSearch])
+
   const assetSelectComponent = useMemo(() => {
     return (
-      <TradeAssetSelect assetId={stakingAsset?.assetId} isReadOnly onlyConnectedChains={true} />
+      <TradeAssetSelect
+        assetId={asset?.assetId}
+        onAssetClick={handleStakingAssetClick}
+        // eslint-disable-next-line react-memo/require-usememo
+        onAssetChange={asset => setAssetId(asset.assetId)}
```

Comment:
can definitely memoize this so no need to disable this linting rule line either

Reply:
824e23a33